services/movie_session.py

```diff
@@ -19,43 +19,44 @@ def create_movie_session(
 
 
 def get_movies_sessions(session_date: str = "") -> MovieSession | None:
+    sessions = MovieSession.objects.all()
     if session_date:
-        year, month, day = session_date.split("-")
-        return MovieSession.objects.filter(
-            show_time__date=date(int(year), int(month), int(day)),
+        sessions = sessions.filter(
+            show_time__date=session_date
         )
-    return MovieSession.objects.all()
+    return sessions
 
 
 def get_movie_session_by_id(movie_id: int) -> MovieSession:
-    return MovieSession.objects.get(id=movie_id)
+    return MovieSession.objects.get(pk=movie_id)
 
 
 def update_movie_session(
         session_id: int,
         show_time: Optional[datetime] = None,
         movie_id: Optional[int] = None,
         cinema_hall_id: Optional[int] = None
-) -> None:
+) -> tuple[Any, Any]:
+    if not any([show_time, movie_id, cinema_hall_id]):
+        return
     try:
-        movie_session = MovieSession.objects.get(id=session_id)
+        movie_session = MovieSession.objects.get(pk=session_id)
     except ObjectDoesNotExist:
-        print(f"Movie Session with ID {session_id} does not exist")
+        raise ObjectDoesNotExist(
+            f"Movie Session with ID {session_id} does not exist"
+        )
     else:
         if show_time:
             movie_session.show_time = show_time
         if cinema_hall_id:
             movie_session.cinema_hall = (
-                CinemaHall.objects.get(id=cinema_hall_id)
+                CinemaHall.objects.get(pk=cinema_hall_id)
             )
         if movie_id:
-            movie_session.movie = Movie.objects.get(id=movie_id)
+            movie_session.movie = Movie.objects.get(pk=movie_id)
         movie_session.save()

```

Comment:
Re-read my last comment about it. The issue isn't in pk, it's because you make redundant queries. Use the fields I mentioned earlier